### PR TITLE
UTM Zone conversion

### PR DESCRIPTION
**Pull request by ateucher**
This adds a function `utm_convert()` to convert tabular data in UTM zones to a standard CRS (default BC Albers). This is especially useful for field studies with data collected over a wide area that spans UTM zones. 

The location data is expected to be in two columns with x and y, and the zone can be either in a column (when the data set spans multiple UTM zones), or can be specified for the whole data set. E.g.:

``` r
# Data with multiple zones, and a column denoting the zone
df <- data.frame(
 animalid = c("a", "b", "c"),
 zone = c(10, 11, 11),
 easting = c(500000, 800000, 700000),
 northing = c(5000000, 3000000, 1000000)
)

utm_convert(df, xcol = "easting", ycol = "northing", zone = "zone")
#> Simple feature collection with 3 features and 6 fields
#> Geometry type: POINT
#> Dimension:     XY
#> Bounding box:  xmin: 1237767 ymin: -3576605 xmax: 2418043 ymax: 21953.96
#> Projected CRS: NAD83 / BC Albers
#>      animalid zone       X           Y easting northing
#> 10          a   10 1237767    21953.96   5e+05    5e+06
#> 11.2        b   11 2275642 -1807950.91   8e+05    3e+06
#> 11.3        c   11 2418043 -3576604.70   7e+05    1e+06
#>                      geometry
#> 10   POINT (1237767 21953.96)
#> 11.2 POINT (2275642 -1807951)
#> 11.3 POINT (2418043 -3576605)

# Data all in one zone, specify a single zone:
df <- data.frame(
 animalid = c("a", "b"),
 easting = c(500000, 800000),
 northing = c(5000000, 3000000)
)

utm_convert(df, xcol = "easting", ycol = "northing", zone = 11)
#> Simple feature collection with 2 features and 5 fields
#> Geometry type: POINT
#> Dimension:     XY
#> Bounding box:  xmin: 1711595 ymin: -1807951 xmax: 2275642 ymax: 62209.93
#> Projected CRS: NAD83 / BC Albers
#>   animalid       X           Y easting northing                 geometry
#> 1        a 1711595    62209.93   5e+05    5e+06 POINT (1711595 62209.93)
#> 2        b 2275642 -1807950.91   8e+05    3e+06 POINT (2275642 -1807951)
```

<sup>Created on 2023-11-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: R/utm-convert.R [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+#'
+#' This can operate on a data frame containing coordinates from multiple UTM zones
+#' with a column denoting the zone, or a single zone for the full dataset.
+#'
+#' It supports data collected in either the NAD83 or WGS84 ellipsoid in
+#' the Northern hemisphere
+#'
+#' @param x data.frame containing UTM coordinates, with a zone column
+#' @param xcol the name of the 'easting' columnm
+#' @param ycol the name of the 'northing' column
```
Comment (boshek) [inline R/utm-convert.R:11]: Any reason why we aren't just calling these `easting_col` and `northing_col`? Seems more direct for a user. 
```suggestion
#' @param xcol the name of the 'easting' column
#' @param ycol the name of the 'northing' column
```

File: R/utm-convert.R [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      utm_zone_lookup[["datum"]] == datum
+    ]
+}
+
+format_zone <- function(x) {
+  if (is.numeric(x)) return(x)
+
+  if (grepl("[Ss]$", x)) {
+    stop("It looks like your UTM zones are in the Southern hemisphere", call. = FALSE)
+  }
```
Comment (boshek) [inline R/utm-convert.R:96]: Fantastic. Very kind.

**Review by boshek — CHANGES_REQUESTED** [review]
Looks good. One comment about argument naming though.

File: R/utm-convert.R (hunk 1 above)
Comment (ateucher) [reply R/utm-convert.R]: Yeah that's probably better. The other option would be to do it the same as `sf` and have one `coords` argument to which you pass a two-element character vector of the column names?

Comment (boshek) [reply R/utm-convert.R]: That could totally work. I always have to pause and remember the order of that vector when I use that function so I'd lean towards separate arguments.

Comment (ateucher) [reply R/utm-convert.R]: Yeah I agree

Comment (ateucher) [reply R/utm-convert.R]: I just went with `"easting"` and `"northing"` for the column argument names

**Review by boshek — APPROVED** [review]
Nice. Great addition. Ship it!

**Pull request merged**